textpipe/wrappers.py

```diff
@@ -0,0 +1,107 @@
+"""
+Wrappers around classes of external libraries used in Doc class.
+"""
+
+import bz2
+import pickle
+from functools import lru_cache
+
+import numpy as np
+from gensim.models.keyedvectors import KeyedVectors
+from redis import Redis
+from redis.exceptions import RedisError
+from tqdm import tqdm
+
+
+class RedisKeyedVectorException(Exception):
+    """ Raised when RedisKeyedVectors class fails"""
+    pass
+
+
+class RedisKeyedVectors(KeyedVectors):
+    """
+    Class to imitate gensim's KeyedVectors, but instead getting the vectors from the memory,
+    the vectors will be retrieved from a redis db.
+    Based on https://engineering.talentpair.com/serving-word-vectors-for-distributed-computations-c5065cbaa02f
+    """
+
+    def __init__(self, host, port, db=0, key='', max_lru_cache_size=1024):
+        self.word_vec = lru_cache(maxsize=max_lru_cache_size)(self.word_vec)
+        self.syn0 = []
+        self.syn0norm = None
+        self.index2word = []
+        self.key = f'w2v_{key}'
+
+        try:
+            self._redis = Redis(host, port, db)
+        except RedisError as e:
+            raise RedisKeyedVectorException(f'The connection to Redis failed while trying to '
+                                            f'initiate the client. Redis error message: {e}')
+
+    @classmethod
+    def load_word2vec_format(cls, **kwargs):
+        raise NotImplementedError(
+            "You can't load a word model that way. It needs to pre-loaded into redis")
+
+    def save(self, *args, **kwargs):
+        raise NotImplementedError("You can't write back to Redis that way.")
+
+    def save_word2vec_format(self, **kwargs):
+        raise NotImplementedError("You can't write back to Redis that way.")
+
+    def word_vec(self, word):
```

Comment:
The super class' `word_vec` method has an additional parameter `use_norm`. Shouldn't that be here too?

Reply:
We don't use that parameter anywhere, so I think we can safely ignore it.

Reply:
Is the intended behavior that only the methods that are overridden in this class are supported on instances of this class? I.e., `RedisKeyedVectors().most_similar()` doesn't work as it described in the super class?

Reply:
Yes that is the intended behavior. This wrapper does not support all methods of its superclass, I only implemented what we use.

Reply:
What is the reason then to extend `KeyedVector`?

Reply:
Good point, for this use case it's not necessary at all (and misleading). I'll remove the inheritance.